sdk/inc/azure/core/az_context.h

```diff
@@ -88,7 +86,8 @@ az_context_create_with_expiration(az_context const* parent, int64_t expiration)
 AZ_NODISCARD AZ_INLINE az_context
 az_context_create_with_value(az_context const* parent, void const* key, void const* value)
 {
-  return (az_context){ ._internal = { .parent = (parent != NULL) ? parent : &az_context_application,
+  _az_PRECONDITION_NOT_NULL(parent);
```

Comment:
Null key isn't allowed.
```suggestion
  _az_PRECONDITION_NOT_NULL(parent);
  _az_PRECONDITION_NOT_NULL(key);
```

Reply:
Fixed